### ISSUE TITLE
`used-before-assignment` false positive with nonlocal

### ISSUE THREAD

**Issue by darintay** [PyCQA/pylint#5965]
### Bug description

Getting a false positive for `used-before-assignment` in the latest pylint release when the variable is declared nonlocal.  I don't think the new try/except logic is handling it correctly, I believe the code below is valid.

```python
# pylint: disable=missing-docstring
def outer():
    myvar = 1
    def inner():
        nonlocal myvar
        try:
            print("A", myvar)
            raise IOError("emyvarc")
        except IOError:
            print("B", myvar)
            myvar += 1
            print("C", myvar)
    inner()
outer()
```

```
test.py:7:23: E0601: Using variable 'myvar' before assignment (used-before-assignment)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
test.py:7:23: E0601: Using variable 'myvar' before assignment (used-before-assignment)
```

### Expected behavior

No pylint errors raised

### Pylint version

```shell
pylint 2.13.0
astroid 2.11.1
Python 3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:21) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 18.04

**Comment by jacobtylerwalls** [comment]
Thanks for the report, regression in #5402.
